File: src/main/java/i/talk/domain/Message.java
package i.talk.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Setter;

@Setter
public class Message {

    private String payload;
    @JsonSerialize(as = Participant.class)
    private Participant sender;

    @JsonProperty("timeStamp")
    private Long timeStamp;

    public Message(String payload, Participant sender) {
        this.payload = payload;
        this.sender = sender;
    }

    public Message() {
    }

    public Message(String payload) {
        this.payload = payload;
    }

    public String getPayload() {
        return payload;
    }

    public void setPayload(String payload) {
        this.payload = payload;
    }

    public Participant getSender() {
        return sender;
    }

    public Long getTimeStamp() {
        return timeStamp;
    }
}

File: frontend/src/app/model/message.ts
import {Person} from "./person";
import {Websocketmessage} from "./websocketmessage";


export class Message implements Websocketmessage {

    payload: string;
    sender: Person;
    timeStamp: number;

    constructor(content: string, sender: Person) {
        this.payload = content;
        this.sender = sender;
    }

}